transaction/core/src/blockchain/block_signature.rs

```diff
@@ -51,10 +49,8 @@ impl BlockSignature {
         block: &Block,
         keypair: &Ed25519Pair,
     ) -> Result<Self, Ed25519SignatureError> {
-        // SHA512 is used for compatibility with Ed25519ph.
```

Comment:
@jcape @mfaulk is there some reason that we need this compatibility? we could use SHA512 here via the pseudo-merlin thing if its important

Reply:
> For Ed25519ph, phflag=1 and PH is SHA512 instead.  That is, the input is hashed using SHA-512 before signing with Ed25519.

https://tools.ietf.org/html/rfc8032

Reply:
this doesn't answer my question -- there's nothing wrong with computing a merlin hash, and then signing that using ed25519. there might be a SHA512 computed on the merlin hash value -- that's fine, efficiency is not a concern

Reply:
You want to set the phflag=1 on Ed25519 in order to distinguish between a signature over X bytes, and a signature over the hash of X bytes. In Ed25519ph, the requirement is that that pre-hash uses SHA512.

Reply:
i see, thanks